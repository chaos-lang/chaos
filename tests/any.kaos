any var9 = 'string5'
var9 = 'string6'
print var9
any var10 = 'string7'
var10 = 123
print var10
any var11 = 'string8'
var11 = true
print var11
any var12 = 'string9'
var12 = 5.6
print var12

any var13 = 123
var13  = 456
print var13
any var14 = 1234
var14  = 'string10'
print var14
any var15 = 12345
var15  = true
print var15
any var16 = 123456
var16  = 1.43
print var16

any var17 = true
var17 = false
print var17
any var18 = false
var18 = 'string11'
print var18
any var19 = true
var19 = 123
print var19
any var20 = false
var20 = 4.1312
print var20

any var21 = 3.14
var21 = 1.44
print var21
any var22 = 3.55
var22 = 'string12'
print var22
any var23 = 6.55
var23 = 1424
print var23
any var24 = 8.55
var24 = true
print var24

any a = 5
num b = 3
print a
print b
a = b
print a

num d = 27
any c = d
print c
print d

print "START OF any to type safe variable assignments"

del a
del b
del c
any a = false
bool b = true
print a
print b
b = a
bool c = a
print a
print b
print c

del a
del b
del c
any a = 0
bool b = true
print a
print b
b = a
bool c = a
print a
print b
print c

del a
del b
del c
any a = 222
bool b = false
print a
print b
b = a
bool c = a
print a
print b
print c

del a
del b
del c
any a = 0.0
bool b = true
print a
print b
b = a
bool c = a
print a
print b
print c

del a
del b
del c
any a = 222.22
bool b = false
print a
print b
b = a
bool c = a
print a
print b
print c

del a
del b
del c
any a = ""
bool b = true
print a
print b
b = a
bool c = a
print a
print b
print c

del a
del b
del c
any a = "asd"
bool b = false
print a
print b
b = a
bool c = a
print a
print b
print c

del a
del b
del c
any a = false
num b = 333
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = true
num b = 333
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = 444
num b = 333
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = 555.55
num b = 333
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = "666"
num b = 333
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = false
num b = 333.33
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = true
num b = 333.33
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = 444
num b = 333.33
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = 555
num b = 333.33
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = "666"
num b = 333.33
print a
print b
b = a
num c = a
print a
print b
print c

del a
del b
del c
any a = false
str b = "hey"
print a
print b
b = a
str c = a
print a
print b
print c

del a
del b
del c
any a = true
str b = "hey"
print a
print b
b = a
str c = a
print a
print b
print c

del a
del b
del c
any a = "universe"
str b = "hey"
print a
print b
b = a
str c = a
print a
print b
print c


del a
del b
del c
any a = 3 + 5
print a
any b = 5 > 3
print b
any c = 0.14 + 3
print c
