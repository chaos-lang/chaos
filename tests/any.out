string6
123
true
5.6
456
string10
true
1.43
false
string11
123
4.1312
1.44
string12
1424
true
5
3
3
27
27
START OF any to type safe variable assignments
false
true
false
false
false
0
true
0
false
false
222
false
222
true
true
0
true
0
false
false
222.22
false
222.22
true
true

true

false
false
asd
false
asd
true
true
false
333
false
0
0
true
333
true
1
1
444
333
444
444
444
555.55
333
555.55
555
555.55
666
333
666
666
666
false
333.33
false
0
0
true
333.33
true
1
1
444
333.33
444
444
444
555
333.33
555
555
555
666
333.33
666
666
666
false
hey
false
false
false
true
hey
true
true
true
333
hey
333
333
333
333.33
hey
333.33
333.33
333.33
universe
hey
universe
universe
universe
8
true
3.14
